drivers/ThinQ_Dryer.groovy

```diff
@@ -150,7 +150,7 @@ def processStateData(data) {
     delayTimeDisplay = parent.convertSecondsToTime(delayTime)
 
     if (parent.checkValue(data,'State')) {
-      String currentStateName = parent.cleanEnumValue(data["State"], "@WM_STATE_")
+      String currentStateName = parent.cleanEnumValue(data["State"], "@WM_STATE_").capitalize()
```

Comment:
This makes things inconsistent with the rest of Hubitat. Attribute values are always lowercase such as "open" and "closed" or "present" and "not present". I don't think it makes sense to make things inconsistent here. I like keeping things consistent with the way Hubitat handles all other devices.

Reply:
Again, my personal **obsessive focus** on providing a custom attribute for display in HE dashboard that looks nice without using Tile# CSS.  You are absolutely correct, **capabilities** attribute "enum" values are always stored in lowercase, except for a few minor exceptions, where the expected value is not a pre-defined list such as a "ENUM"  but defined as a "STRING" like trackDescription in a mediaController, or the various STRING values for the TV capability, etc...

I personally believe that custom attribute values in devices can be stored in any case, uppercase, lowercase, titlecase, mixed case, etc depending on the developers style and conventions, which I totally respect yours is to  'lowercase' and follows the normal standard capabilities ENUM convention...    

In HE's dashboard, many device states of HE's native **device capabilities** are auto capitalized for display in the dashboard app, for example, "On/Off", "Closed,Open", "Modes", "Present/Not Present", "Active/Inactive", etc...  However, you are totally right, a device's capabilities are stored in lowercase, so the HE dashboard app, dynamically converts these capabilities attribute values to a capitalize value for dashboard display...Certainly there are many ways to skin a cat here, including using RM's global variables in the Dashboard to convert attribute strings to whatever the user desires...  

This proposed change to an established custom attribute, that your users base might already code fo or like being lowercase, should remain lowercase.  I can "pretty it up" with another new attribute "currentStateNameDisplay" in my fork for dashboard display.

Again thanks for all you do in this excellent app that I appreciate....

Reply:
Yeah I get your point. I guess I'm a purist the other way, I HATE that device drivers have to add these stupid duplicate ":display" attributes (which take up space in the DB, etc.) because of limitations of the dashboard! I have so many ideas in my head of how I'd improve that dashboard! In my mind, you should have something like:

`attribute "myAttr", "string", method_to_format_for_dashboard_use` and then you can do:

`def method_to_format_for_dashboard_use(value) { return value+ getTemperatureScale() }`

Or something along those lines, but I digress.

For me personally I use the states to trigger automations so this would break my stuff. Like I have a "dishwasher is done" run that checks the current state. 

So I'm thinking of two possibilities:
1.) Add a setting in the app for capitalize and it would be applied consistently across all the drivers in this app (fyi, personally I'd prefer title case over capitalize, I think "Power Off" looks better than "Power off" but that's just me!
2.) Add duplicate attributes like you mentioned for currentStateNameDisplay

I've kind of always held off on going crazy adding all this stuff to my drivers in the hopes that the HE team would make the dashboard better, but it seems like the dashboard is the ugly step child that no one wants to go near!

I'm going to merge in the other changes today so we have working support for v1 fridges and I'll think a bit more on this one. Which of the above two solutions do you think is better?

Reply:
I'm with you on the disliking the multitude of variations to support a pretty dashboard display.  

For example, my Ambient Weather App is coded to allow user preferences for displaying their data values as imperial, metric, and difference units of measure, # of decimals, etc..  For example, here is Wind Speed and their are several other fields:
```
input ( name: "windUnits", type: "enum",
                   title: "Select Wind Speed Units of Measure",
                   options: ['mph':'Miles per Hour','fps':'Feet per Second','mps':'Meter per Second','kph':'Kilometers per Hour','knotts':'Knotts'],
                   defaultValue: "mph",
                   required: true
                  )
```

I have had users located in the USA, which I had assumed they would like Imperial units and that turned out to be false, they liked metric because they either just moved here from Europe/Asia, or that they wanted to mix & match measurements...  so hundreds of extra lines of logic coding to please the masses....  

Luckily, in ThinQ, this extra code is only to accomodate a "STRING" status format for the Dashboard or Pushover messages.  I agree with you that **Titlecase** would be an easy fix for a new field maybe named something like 'currentStateDashboard'.  Maybe a function like this? 

```
def titleCase(value) { 
      value.split("\\s+").collect { it.toLowerCase().capitalize() 
}
```

I would not implement my initial recommendation for adding extra fields in the ThinQ_Fridge for the units of temperature measurement for refrig and freeze fields.  In my V1, these temp values are the set temps and not actual inside temp measurements that would be more valuable for home automation.